tests/test_uri_templates.py

```diff
@@ -164,6 +166,43 @@ def test_converter_rejections(client, uri_template):
     assert not resource1.called
 
 
+@pytest.mark.parametrize('uri_template, path, dt_expected', [
+    (
+        '/{start_year:int}-to-{timestamp:dt}',
+        '/1961-to-1969-07-21T02:56:00Z',
+        datetime(1969, 7, 21, 2, 56, 0)
+    ),
+    (
+        '/{start_year:int}-to-{timestamp:dt("%Y-%m-%d")}',
+        '/1961-to-1969-07-21',
+        datetime(1969, 7, 21)
+    ),
+    (
+        '/{start_year:int}/{timestamp:dt("%Y-%m-%d %H:%M")}',
+        '/1961/1969-07-21 14:30',
+        datetime(1969, 7, 21, 14, 30)
+    ),
+    (
+        '/{start_year:int}-to-{timestamp:dt("%Y-%m")}',
+        '/1961-to-1969-07-21',
+        None
+    ),
+])
+def test_datetime_converter(client, resource, uri_template, path, dt_expected):
+    client.app.add_route(uri_template, resource)
+
+    result = client.simulate_get(path)
+
+    if dt_expected is None:
+        assert result.status_code == 404
+        assert not resource.called
+    else:
+        assert result.status_code == 200
+        assert resource.called
+        assert resource.captured_kwargs['start_year'] == 1961
+        assert resource.captured_kwargs['timestamp'] == dt_expected
```

Comment:
👍 for moving the None use case into its own test case

I now remember I did something similar at work with a group of related use cases....I should break them out into separate tests...yay for learning via open source! 😄